services/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/builders/HuaweiImageBuilder.java

```diff
@@ -67,4 +80,93 @@ public boolean destroy(BuilderContext ctx) {
         // TODO: destroy the temporary images here or after resources created.
         return true;
     }
+
+    protected TFExecutor prepareExecutor(BuilderContext ctx) {
+        if (ctx == null) {
+            log.error("BuilderContext is invalid.");
+            throw new BuilderException(this, "Builder context is null.");
+        }
+        Map<String, String> envCtx = ctx.get(new HuaweiEnvBuilder(ocl).name());
+        if (envCtx == null) {
+            log.error("Dependent builder: {} must build first.", new HuaweiEnvBuilder(ocl).name());
+            throw new BuilderException(this, "HuaweiEnvBuilder context is null.");
+        }
+        TFExecutor executor = new TFExecutor(envCtx);
+        executor.createWorkspace(name());
+        executor.createTFScript(""
+            + "resource \"huaweicloud_vpc\" \"vpc\" {\n"
+            + "  name = \"osc-packer-01\"\n"
+            + "  cidr = \"192.168.0.0/16\"\n"
+            + "}\n"
+            + "\n"
+            + "resource \"huaweicloud_vpc_subnet\" \"subnet\" {\n"
+            + "  name       = \"osc-packer-01\"\n"
+            + "  cidr       = \"192.168.1.0/24\"\n"
+            + "  gateway_ip = \"192.168.1.1\"\n"
+            + "  vpc_id     = huaweicloud_vpc.vpc.id\n"
+            + "}\n"
+            + "\n"
+            + "resource \"huaweicloud_networking_secgroup\" \"secgroup\" {\n"
+            + "  name        = \"osc-packer-secgroup_1\"\n"
+            + "  description = \"My security group\"\n"
+            + "}\n"
+            + "\n"
+            + "resource \"huaweicloud_networking_secgroup_rule\" \"test\" {\n"
+            + "  security_group_id = huaweicloud_networking_secgroup.secgroup.id\n"
+            + "  direction         = \"ingress\"\n"
+            + "  ethertype         = \"IPv4\"\n"
+            + "  protocol          = \"tcp\"\n"
+            + "  port_range_min    = 22\n"
+            + "  port_range_max    = 22\n"
+            + "  remote_ip_prefix  = \"159.138.32.195/32\"\n"
+            + "}");
+
+        return executor;
+    }
+
+    public PackerVars prepareEnv(TFExecutor tfExecutor) {
+
+        PackerVars packerVars = new PackerVars();
+        log.info("Creating Huawei Cloud resources.");
+
+        tfExecutor.tfInit();
+        tfExecutor.tfPlan();
+        tfExecutor.tfApply();
+        String stateContent = tfExecutor.getTFState();
+
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
ObjectMapper is a heavy weight object, how about reusing it across execution ?

Reply:
ok, let me do it.